marketplace_standard_app_api/main.py

```diff
@@ -0,0 +1,245 @@
+        500: {"description": "Internal server error."},
+        501: {"description": "Not implemented."},
+        503: {"description": "Service unavailable."},
+    },
+)
+async def get_dataset(dataset_id: DatasetId) -> DatasetModel:
+    """Retrieve an existing data set."""
+    raise HTTPException(status_code=501, detail="Not implemented.")
+
+
+@api.post(
+    "/transformations",
+    operation_id="newTransformation",
+    tags=["Transformation"],
+    response_model=TransformationCreateResponse,
+    responses={
+        401: {"description": "Not authenticated."},
+        500: {"description": "Internal server error."},
+        501: {"description": "Not implemented."},
+        503: {"description": "Service unavailable."},
+    },
+)
+async def create_transformation(
+    transformation: NewTransformationModel,
+) -> TransformationCreateResponse:
+    """Create a new transformation."""
+    raise HTTPException(status_code=501, detail="Not implemented.")
+
+
+@api.get(
+    "/transformations/{transformation_id}",
+    operation_id="getTransformation",
+    tags=["Transformation"],
+    response_model=TransformationModel,
+    responses={
+        401: {"description": "Not authenticated."},
+        404: {"description": "Not found."},
+        500: {"description": "Internal server error."},
+        501: {"description": "Not implemented."},
+        503: {"description": "Service unavailable."},
+    },
+)
+async def get_transformation(
+    transformation_id: TransformationId,
+) -> TransformationModel:
+    """Retrieve an existing transformation."""
+    raise HTTPException(status_code=501, detail="Not implemented.")
+
+
+@api.delete(
+    "/transformations/{transformation_id}",
+    operation_id="deleteTransformation",
+    tags=["Transformation"],
+    status_code=204,
+    responses={
+        401: {"description": "Not authenticated."},
+        404: {"description": "Not found."},
+        500: {"description": "Internal server error."},
+        501: {"description": "Not implemented."},
+        503: {"description": "Service unavailable."},
+    },
+)
+async def delete_transformation(
+    transformation_id: TransformationId,
+) -> Optional[HTTPException]:
+    """Delete an existing transformation."""
+    raise HTTPException(status_code=501, detail="Not implemented.")
+
+
+@api.patch(
+    "/transformations/{transformation_id}",
+    operation_id="updateTransformation",
+    tags=["Transformation"],
+    response_model=TransformationUpdateResponse,
+    responses={
+        401: {"description": "Not authenticated."},
+        404: {"description": "Not found."},
+        409: {
+            "description": "The requested state is unavailable (example: trying to stop an already completed transformation)."
+        },
+        500: {"description": "Internal server error."},
+        501: {"description": "Not implemented."},
+        503: {"description": "Service unavailable."},
+    },
+)
+async def update_transformation(
```

Comment:
- [ ] Expand documentation on `updateTransformation` operation.

Reply:
@pablo-de-andres @yoavnash While revising the docs on the transformations resource, I realized that we might want to adopt the term "state" rather than "status" to better describe what we are trying to convey. Our transformation goes through multiple states (CREATED -> RUNNING -> STOPPED). The final outcome could be described as a status (COMPLETED / FAILED). However, the latter two could also be described as "_states_".

We are usually interested in the transformation _state_, not the transformation _status_. Note also that I use the term "_states_" to describe a plurality of _status_ which sounds much better than _statuses_ in my opinion.

Additional references:
_Note that these are actually contradictory and I do not think there is a "correct" answer to this question._

- https://stackoverflow.com/a/1162838
- https://softwareengineering.stackexchange.com/a/219368

Reply:
After offline discussion we have decided to move forward with *state / states*.